modules/library/main/src/main/java/org/geotools/gml/producer/CoordinateFormatter.java

```diff
@@ -0,0 +1,129 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2018, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geotools.gml.producer;
+
+import java.text.FieldPosition;
+import java.text.NumberFormat;
+import java.util.Locale;
+
+/**
+ * Formats coordinates with a given number of decimals, using code more efficient than NumberFormat when possible
+ */
+public class CoordinateFormatter {
+
+    /**
+     * The min value at which the decimal notation is used 
+     * (below it, the computerized scientific one is used instead)
+     */
+    private static final double DECIMAL_MIN = Math.pow(10, -3);
+
+    /**
+     * The max value at which the decimal notation is used 
+     * (above it, the computerized scientific one is used instead)
+     */
+    private static final double DECIMAL_MAX = Math.pow(10, 7);
+
+    /** To be used for formatting numbers, uses US locale. */
+    private final NumberFormat coordFormatter = NumberFormat.getInstance(Locale.US);
+
+    private final FieldPosition ZERO = new FieldPosition(0);
+
+    /**
+     * The power of ten used for fast rounding
+     */
+    private final double scale;
+
+    /**
+     * Whether we have to format in plain decimal numbers, or we can use scientific notation
+     */
+    private boolean forcedDecimal;
+
+
+    public CoordinateFormatter(int numDecimals) {
+        coordFormatter.setMaximumFractionDigits(numDecimals);
+        coordFormatter.setGroupingUsed(false);
+        scale = Math.pow(10, numDecimals);
+    }
+
+    /**
+     * Formats a number with the configured number of decimals. For better performance best use 
+     * {@link #format(double, StringBuffer)} against a re-used StringBuffer
+     * @param x
+     * @param sb
+     */
+    public String format(double x) {
+        StringBuffer sb = new StringBuffer();
+        format(x, sb);
+        return sb.toString();
+    }
+
+    /**
+     * Formats a number with the configured number of decimals
+     * @param x
+     * @param sb
+     */
+    public StringBuffer format(double x, StringBuffer sb) {
+        if ((Math.abs(x) >= DECIMAL_MIN && x < DECIMAL_MAX) || x == 0) {
+            x = truncate(x);
+            long lx = (long) x;
+            if (lx == x)
+                sb.append(lx);
+            else
+                sb.append(x);
```

Comment:
I would substitute:
```
if (lx == x)
  sb.append(lx);
else
  sb.append(x);
```

with:

```
sb.append(lx == x ? lx : x);
```

or add the missing ``{}``.

Reply:
Braces added